### PR TITLE
Feacture/pending requirements

### PR DESCRIPTION
**Pull request by lrojas870**
En este pull request se agregaron las siguiente tareas segun el board del proyecto:

- Agregar una primera pantalla como splash con la imagen adjunta
- Agregar un loading a la pantalla home
- Crear date picker
- Integrar date picker con funcionalidad que busca la informacion
- Agregar validación en la fecha seleccionada en el date picker

### CODE REVIEW

File: app/src/main/java/com/androidavanzado/covid19stadistics/util/MonthToNameMonth.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.androidavanzado.covid19stadistics.util
+
+class MonthToNameMonth {
+
+    operator fun invoke(month: String) : String {
+
+        val nameMonth = when (month) {
```
Comment (iMoya917) [inline app/src/main/java/com/androidavanzado/covid19stadistics/util/MonthToNameMonth.kt]: You could use the date formats and not have a when to know the name of the month.

File: app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt [hunk 2]
```diff
@@ -13,17 +13,15 @@ class RemoteDataSource(private val apiClient: ApiClient) : DataSource {
...
 
-                Log.e("XXXXX","${it.isSuccessful}")
-
                 return if(it.isSuccessful && it.body()!=null) {
                     val data = it.body()?.data
                     Result.Success(data)
                 }
                 else{
-                    Result.Failure("Sin datos")
+                    Result.Failure("No se encontraron datos para la fecha seleccionada.")
```
Comment (manuel28g-carsales) [inline app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt]: Try don't use a hardcoded string, instead use the resource file

File: app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt [hunk 3]
```diff
@@ -13,17 +13,15 @@ class RemoteDataSource(private val apiClient: ApiClient) : DataSource {
...
                 }
             }?:run{
-                return Result.Failure("Ocurrió un error")
+                return Result.Failure("Ocurrió un error, tratanto de obtener la información.")
```
Comment (manuel28g-carsales) [inline app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt]: Same here, use the res/values/strings file for this cases

File: app/src/main/java/com/androidavanzado/covid19stadistics/ui/DatePickerFragment.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    override fun onDateSet(view: DatePicker?, year: Int, month: Int, dayOfMonth: Int) {
+
+        val newMonth = month + 1
+        listener(dayOfMonth, newMonth, year)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+
+        val c = Calendar.getInstance()
```
Comment (manuel28g-carsales) [inline app/src/main/java/com/androidavanzado/covid19stadistics/ui/DatePickerFragment.kt]: change this variable name to another one more verbose like "calendar"

File: app/src/main/java/com/androidavanzado/covid19stadistics/util/DateTextFormat.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import android.annotation.SuppressLint
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.Date
+
+class DateTextFormat(val monthToNameMonth : MonthToNameMonth) {
+
+    @SuppressLint("SimpleDateFormat")
+    operator fun invoke(date: Date): String {
+        val dayFormat: DateFormat = SimpleDateFormat("dd")
```
Comment (manuel28g-carsales) [inline app/src/main/java/com/androidavanzado/covid19stadistics/util/DateTextFormat.kt]: This assignment of SimpleDateFormat should be in a global scope to be created once and not each time when call invoke method

File: app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            if(monthString.length<2){
+                monthString = "0$monthString"
+            }
+
+            val dateString = "$year-$monthString-$day"
+
+            callGetData(dateString)
+        }
+        else{
+            Toast.makeText(this, "No puede seleccionar una fecha igual o mayor a hoy.", Toast.LENGTH_SHORT).show()
```
Comment (manuel28g-carsales) [inline app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt]: Move all the string of this file to the res/values/string source

File: app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt (hunk 2 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt]: done

File: app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt (hunk 3 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/storage/remote/RemoteDataSource.kt]: done

File: app/src/main/java/com/androidavanzado/covid19stadistics/ui/DatePickerFragment.kt (hunk 4 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/ui/DatePickerFragment.kt]: done

File: app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt (hunk 6 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt]: done

File: app/src/main/java/com/androidavanzado/covid19stadistics/util/MonthToNameMonth.kt (hunk 1 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/util/MonthToNameMonth.kt]: done

File: app/src/main/java/com/androidavanzado/covid19stadistics/util/DateTextFormat.kt (hunk 5 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/util/DateTextFormat.kt]: done

**Review by iMoya917 — APPROVED** [review]

**Pull request merged**